src/tracers.jl

```diff
@@ -240,6 +219,10 @@ end
 # Utilities #
 #===========#
 
+myempty(::Type{ConnectivityTracer{P}}) where {P} = ConnectivityTracer{P}(myempty(P), true)
+myempty(::Type{GradientTracer{P}}) where {P} = GradientTracer{P}(myempty(P), true)
+myempty(::Type{HessianTracer{P}}) where {P} = HessianTracer{P}(myempty(P), true)
```

Comment:
In the test suite, we want to check that this is allocation-free. I am not that far but I assume we have the same overloads for `myempty` on common set and element types so that it points to a constant empty set?

Reply:
Nope, constant empty sets were removed in #57 without losing notable amounts of performance. We can try reintroducing them in a follow-up, now that we have benchmarks?